samples/core/xgboost_training_cm/xgboost_training_cm.py

```diff
@@ -201,86 +226,94 @@ def dataproc_predict_op(
+    analyze_output = output_template
+    transform_output_train = os.path.join(output_template, 'train', 'part-*')
+    transform_output_eval = os.path.join(output_template, 'eval', 'part-*')
+    train_output = os.path.join(output_template, 'train_output')
+    predict_output = os.path.join(output_template, 'predict_output')
+
+    with dsl.ExitHandler(exit_op=dataproc_delete_cluster_op(
+        project_id=project,
+        region=region,
+        name=cluster_name
+    ).apply(gcp.use_gcp_secret('user-gcp-sa'))):
+
+        # create_cluster_op = dataproc_create_cluster_op(
+        #     project=project,
+        #     region=region,
+        #     staging=output
+        # ).apply(gcp.use_gcp_secret('user-gcp-sa'))
 
-    with dsl.ExitHandler(exit_op=delete_cluster_op):
         create_cluster_op = dataproc_create_cluster_op(
-            project,
-            region,
-            output
+            project_id=project,
+            region=region,
+            name=cluster_name,
+            initialization_actions=[
+              os.path.join(_PYSRC_PREFIX,
+                           'create/initialization_actions.sh'),
+            ],
+            image_version='1.2'
         ).apply(gcp.use_gcp_secret('user-gcp-sa'))
 
         analyze_op = dataproc_analyze_op(
-            project,
-            region,
-            create_cluster_op.output,
-            schema,
-            train_data,
-            output_template
-        ).apply(gcp.use_gcp_secret('user-gcp-sa'))
+            project=project,
+            region=region,
+            cluster_name=cluster_name,
+            schema=schema,
+            train_data=train_data,
+            output=output_template
+        ).after(create_cluster_op).apply(gcp.use_gcp_secret('user-gcp-sa'))
 
         transform_op = dataproc_transform_op(
-            project,
-            region,
-            create_cluster_op.output,
-            train_data,
-            eval_data,
-            target,
-            analyze_op.output,
-            output_template
-        ).apply(gcp.use_gcp_secret('user-gcp-sa'))
+            project=project,
+            region=region,
+            cluster_name=cluster_name,
+            train_data=train_data,
+            eval_data=eval_data,
+            target=target,
+            analysis=analyze_output,
+            output=output_template
+        ).after(analyze_op).apply(gcp.use_gcp_secret('user-gcp-sa'))
 
         train_op = dataproc_train_op(
-            project,
-            region,
-            create_cluster_op.output,
-            transform_op.outputs['train'],
-            transform_op.outputs['eval'],
-            target,
-            analyze_op.output,
-            workers,
-            rounds,
-            output_template
-        ).apply(gcp.use_gcp_secret('user-gcp-sa'))
+            project=project,
+            region=region,
+            cluster_name=cluster_name,
+            train_data=transform_output_train,
```

Comment:
You might probably want something like `train_data=str(transform_op.output) + '/train/part-*',` instead.

Reply:
Wish I could... but the current GCP components do not support customized output. That's why the paths are being passed around as strings.

Reply:
> the current GCP components do not support customized output

This seems to be a substantial regression.
They should at least support some generic pass-through output like "Output dir", that they pass through and also try to pass to the user code.

Reply:
> > the current GCP components do not support customized output
> 
> This seems to be a substantial regression.
> They should at least support some generic pass-through output like "Output dir", that they pass through and also try to pass to the user code.

Agreed. That being said, provided that Unified DSL components are already within our planning items, how much efforts shall we put here to enhance the feature of those components?